src/generator.cs

```diff
@@ -7051,3 +7092,388 @@ string RenderType (Type t)
 	}
 	
 }
+
+public static class TypeManager {
```

Comment:
That file is already **huge**, let's move the new type to a `generator-typemanager.cs` file.